core/core/src/sf_core/json_schema_validator.rs

```diff
@@ -36,13 +66,15 @@ impl JsonSchemaValidator {
         let result = self.compiled.validate(&instance);
 
         if let Err(errors) = result {
-            for error in errors {
-                return Err(JsonSchemaValidatorError::ValidationError {
-                    kind: format!("{:?}", error.kind), // TODO: string or keep original?
-                    value: format!("{:?}", error.instance), // TODO: Serde::Value to HostValue?
-                    path: error.instance_path.to_string(),
-                });
-            }
+            return Err(JsonSchemaValidatorError::ValidationErrors(
+                errors
+                    .map(|error| ValidationError {
+                        kind: format!("{:?}", error.kind), // TODO: string or keep original?
+                        value: format!("{:?}", error.instance), // TODO: Serde::Value to HostValue?
```

Comment:
idea: I would consider if it wouldn't produce a prettier error for the user to stringify the JSON here instead of printing it using Debug, but I think we should return to that later if at all and not block on it now.